pkg/splunk/enterprise/clustermaster.go

```diff
@@ -71,6 +71,16 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterprisev1.Clus
 		cr.Status.SmartStore = cr.Spec.SmartStore
 	}
 
+	// ToDo sgontla: Handle
+	// 1. Spec update
+	// 2. S3 credentails change
+	// 3. Probe times expired
+	if !reflect.DeepEqual(cr.Status.AppContext.AppFrameworkConfig, cr.Spec.AppFrameworkConfig) {
+		// TBD: Probe the remote store, and handle any changes on remote store
+
+		cr.Status.AppContext.AppFrameworkConfig = cr.Spec.AppFrameworkConfig
+	}
+
 	// This is to take care of case where AreRemoteVolumeKeysChanged returns an error if it returns false.
```

Comment:
Err check block related to AreRemoteVolumeKeysChanged should preferably be moved prior to the AppFrameworkConfig spec check

Reply:
Addressed